### DISCUSSION TITLE
Custom models in Pimcore 10.0.3

### DISCUSSION THREAD

**Discussion by Mirogniewa** [pimcore/pimcore#9481] [Q&A / Support] [answered]
Hello,

I am new to Pimcore, exploring its possibilities. I've worked with Magento for years, but I do not really have experience with Symfony. 
I've installed Pimcore 10.0.3, it's working fine and now I am trying to save additional data in custom table on object creation/update/delete. I have already created an EventListener which works - reacts when an object is updated via admin panel. My goal is to: 
1. create a custom table to store additional data 
2. create classes to handle communication with database

Question 1: I have created a table in database, but I did it manually. Does Pimcore provide a way of automating this task, so that when we move the project to a different environment, we don't need to remember about all actions I needed to execute on my local environment? In Magento I could create installers which would be assigned a version number and would be executed - or not - in the new environment, depending on settings. 

Question 2: Is there a tool which generates code needed for communication with my custom table (classes with getters, setters, etc...) or do I need to write it all manually? I've tried `pimcore:generate:bundle` command, but it didn't create them. Writing all classes manually would be time consuming.

Thanks in advance!
Sylwia

**Comment by NiklasBr** [comment, marked as answer]
Check out the [Custom Persistent Models chapter in the docs](https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Custom_Persistent_Models.html)

For the table creation you could create a [migration that runs on deploy](https://pimcore.com/docs/pimcore/current/Development_Documentation/Development_Tools_and_Details/Migrations.html) or do it on [install](https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Bundle_Developers_Guide/Pimcore_Bundles/Installers.html)

**Reply by Mirogniewa** [reply]
I've checked what you recommended and successfully created both classes structure and installer for my bundle. Thanks for your reply!
